scripts/BumpVersion.swift

```diff
@@ -0,0 +1,169 @@
+import Foundation
+
+let greenColor = "\u{001B}[0;32m"
+let resetColor = "\u{001B}[0;0m"
+let redColor = "\u{001B}[0;31m"
+let yellowColor = "\u{001B}[33m"
+
+guard CommandLine.arguments.count == 2 else {
+    print("\(redColor)Please provide a version in the format `major.minor.patch`\(resetColor)")
+    exit(1)
+}
+
+func isVersionBumped(_ currentVersion: String, _ newVersion: String) -> Bool {
+    let currentComponents = currentVersion.components(separatedBy: ".")
+    let newComponents = newVersion.components(separatedBy: ".")
+
+    guard currentComponents.count == 3 && newComponents.count == 3 else {
+        return false
+    }
+
+    let currentMajor = Int(currentComponents[0]) ?? 0
+    let currentMinor = Int(currentComponents[1]) ?? 0
+    let currentPatch = Int(currentComponents[2]) ?? 0
+
+    let newMajor = Int(newComponents[0]) ?? 0
+    let newMinor = Int(newComponents[1]) ?? 0
+    let newPatch = Int(newComponents[2]) ?? 0
+
+    if newMajor < currentMajor {
+        return false
+    }
+
+    if newMajor == currentMajor && newMinor < currentMinor {
+        return false
+    }
+
+    if newMajor - currentMajor > 1 {
+        print("\(yellowColor)Warning: New version major number should not jump by more than 1.\(redColor)")
+    }
+
+    if newMajor > currentMajor  {
+        if newMinor != 0 {
+            print("\(yellowColor)Warning: New major version should have a minor number of 0.\(resetColor)")
```

Comment:
Why can't the new major version have a minor version of 0? Do we not want to be able to go to `1.0.0`?